synapse/storage/databases/main/user_directory.py

```diff
@@ -517,18 +516,21 @@ def _filter_local_users_for_dir_txn(
             and not self.get_if_app_services_interested_in_user(user)  # type: ignore[attr-defined]
         ]
 
-        rows = self.db_pool.simple_select_many_txn(
-            txn,
-            table="users",
-            column="name",
-            iterable=users,
-            keyvalues={
-                "deactivated": 0,
-            },
-            retcols=("name", "user_type"),
+        rows = cast(
+            List[Tuple[str, str]],
```

Comment:
user_type is nullable in the DB. Indeed, I think most users will have no user type set?